xdp-dump/xdpdump.c

```diff
@@ -250,24 +365,45 @@ static enum bpf_perf_event_ret handle_perf_event(void *private_data,
 			if (ctx->cfg->pcap_file[0] == '-' &&
 			    ctx->cfg->pcap_file[1] == 0)
 				pcap_dump_flush(ctx->pcap_dumper);
+
+		} else if (ctx->pcapng_dumper) {
+			int64_t  action = e->metadata.action;
+			uint32_t queue = e->metadata.rx_queue;
+			uint64_t pktid = ctx->cpu_packet_id[cpu];
+
+			xpcapng_dump_enhanced_pkt(ctx->pcapng_dumper,
+						  fexit ? 1 : 0,
+						  PCAPNG_EPB_FLAG_INBOUND,
+						  ts, e->metadata.pkt_len,
+						  min(e->metadata.cap_len,
+						      ctx->cfg->snaplen),
+						  e->packet,
+						  ctx->last_missed_events,
+						  NULL, &pktid, &queue,
+						  fexit ? &action : NULL);
```

Comment:
This is quite unreadable... See also previous comment on the definition of this function.

Reply:
Yes the above change will fix this...